packages/redis_client/test/src/redis_client_test.dart

```diff
@@ -1,10 +1,157 @@
+import 'dart:io';
+
 import 'package:redis_client/redis_client.dart';
 import 'package:test/test.dart';
 
 void main() {
   group(RedisClient, () {
-    test('can be instantiated', () {
-      expect(const RedisClient(), isNotNull);
+    late RedisClient client;
+
+    setUp(() async {
+      client = RedisClient();
+      await client.connect();
+    });
+
+    tearDown(() async {
+      try {
+        await client.sendCommand(['RESET']);
+        await client.sendCommand(['FLUSHALL']);
+        await client.close();
+      } catch (_) {
+        // ignore
+      }
+    });
+
+    test('throws SocketException when connection times out', () async {
+      final client = RedisClient(
+        socket: const RedisSocketOptions(timeout: Duration(microseconds: 1)),
+      );
+      await expectLater(
+        () => client.connect(maxConnectionAttempts: 1),
+        throwsA(
+          isA<SocketException>().having(
+            (e) => e.message,
+            'message',
+            contains('Connection timed out'),
+          ),
+        ),
+      );
+    });
+
+    test('throws SocketException after max connection attempts', () async {
+      final client = RedisClient(
+        socket: const RedisSocketOptions(port: 1234),
+      );
+      await expectLater(
+        () => client.connect(maxConnectionAttempts: 1),
+        throwsA(
+          isA<SocketException>().having(
+            (e) => e.message,
+            'message',
+            contains('Connection refused'),
+          ),
+        ),
+      );
+    });
+
+    test('throws SocketException after disconnect', () async {
+      final client = RedisClient(
+        socket: const RedisSocketOptions(port: 1234),
+      );
+      await expectLater(
+        () => client.connect(maxConnectionAttempts: 0),
+        throwsA(
+          isA<SocketException>().having(
+            (e) => '$e',
+            'message',
+            contains('Connection retry limit exceeded'),
+          ),
+        ),
+      );
+    });
+
+    test('throws StateError when closed', () async {
+      await client.close();
+      await expectLater(
+        client.connect(),
+        throwsA(
+          isA<StateError>().having(
+            (e) => e.message,
+            'message',
+            'RedisClient has been closed.',
+          ),
+        ),
+      );
+    });
```

Comment:
I'd consider putting these in a "connection" group

Reply:
yup good call 👍